lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -1989,6 +1990,20 @@ SwiftASTContext::CreateInstance(lldb::LanguageType language, Module &module,
   {
     LLDB_SCOPED_TIMERF("%s (getStdlibModule)", m_description.c_str());
     const bool can_create = true;
+
+    // Report progress on module importing by using a callback function in swift::ASTContext
+    Progress progress("Importing Swift standard library modules");
+    swift_ast_sp->m_ast_context_ap->SetPreModuleImportCallback(
+      [&progress](llvm::StringRef module_name, bool is_overlay) {
+        progress.Increment(1, (is_overlay ? module_name.str() + " (overlay)" : module_name.str()));
+      }
+);
+
+    // Clear the callback function on scope exit to prevent an out-of-scope access of the progress local variable
+    auto on_exit = llvm::make_scope_exit([&](){
+      swift_ast_sp->m_ast_context_ap->SetPreModuleImportCallback([](llvm::StringRef module_name, bool is_overlay) {});
```

Comment:
Does `swift_ast_sp->m_ast_context_ap->SetPreModuleImportCallback({})` work?

Reply:
I tried it locally and it does, but per [Augusto's comment](https://github.com/apple/llvm-project/pull/7769#discussion_r1396471773) we might want to set the callback function to a generic progress report in the event that a module import is triggered outside of these functions.